boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -0,0 +1,127 @@
+BUNDLE_DIR="$SAAS_OPERATOR_DIR/${operator_name}/"
+
+if [ "$diff_generate" = true ] ; then
+    OPERATOR_NEW_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    OPERATOR_PREV_VERSION=$(ls "${BUNDLE_DIR}" | sort -t . -k 3 -g | tail -n 2 | head -n 1)
+    OUTPUT_DIR="output-comparison"
+    
+    # For diff usecase, checking there is already a generated CSV
+    if [ ! -f ${BUNDLE_DIR}/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml ] ; then
+        echo "You need to generate CSV with your legacy script before trying to run the diff option"
+        exit 1 
+    fi
+else
+    rm -rf "$SAAS_OPERATOR_DIR"
+    git clone --branch "$operator_channel" ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+    
+    # remove any versions more recent than deployed hash
+    if [[ "$operator_channel" == "production" ]]; then
+        if [ -z "$DEPLOYED_HASH" ] ; then
+            DEPLOYED_HASH=$(
+                curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml" | \
+                    docker run --rm -i quay.io/app-sre/yq yq r - "resourceTemplates[*].targets(namespace.\$ref==/services/osd-operators/namespaces/hivep01ue1/${_OPERATOR_NAME}.yml).ref"
+            )
+        fi
+    
+        delete=false
+        # Sort based on commit number
+        for version in $(ls $BUNDLE_DIR | sort -t . -k 3 -g); do
+            # skip if not directory
+            [ -d "$BUNDLE_DIR/$version" ] || continue
+    
+            if [[ "$delete" == false ]]; then
+                short_hash=$(echo "$version" | cut -d- -f2)
+    
+                if [[ "$DEPLOYED_HASH" == "${short_hash}"* ]]; then
+                    delete=true
+                fi
+            else
+                rm -rf "${BUNDLE_DIR:?BUNDLE_DIR var not set}/$version"
+            fi
+        done
+    fi
+    # TODO : Clean handling of major version (should be variable from consumer repo and default to 0.1 is undefined)
+    OPERATOR_PREV_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    OPERATOR_NEW_VERSION="0.1.${operator_commit_number}-${operator_commit_hash}"
+    OUTPUT_DIR=${BUNDLE_DIR}
+fi
+
+if [[ "$generate_script" = "common" ]] ; then
+    ./boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py -o ${operator_name} -d ${OUTPUT_DIR} -p ${OPERATOR_PREV_VERSION} -n ${operator_commit_number} -c ${operator_commit_hash} -i ${operator_image}
+elif [[ "$generate_script" = "hack" ]] ; then
+    if [ -z "$OPERATOR_PREV_VERSION" ] ; then 
+        OPERATOR_PREV_VERSION="no-version"
+        DELETE_REPLACE=true
+    fi
+    
+    ./hack/generate-operator-bundle.py ${OUTPUT_DIR} ${OPERATOR_PREV_VERSION} ${operator_commit_number} ${operator_commit_hash} ${operator_image}
+    
+    if [ ! -z "${DELETE_REPLACE}" ] ; then
+        yq d -i output-comparison/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml 'spec.replaces'
```

Comment:
FYI we've had trouble with different versions of `yq` being present wherever this runs. v3 and v4 have wildly different and incompatible syntax.

But if we detect this condition in generate-operator-bundle.py and delete `spec.replaces` there as suggested above, this chunk goes away and we don't have to worry about it :)

Reply:
Objective was to avoid having logic exclusive to the common version which couldn't be supported for the 'hack' scripts. However, if we consider any new operator should use the common version, we could move it to the python script indeed.

Reply:
If you want to keep yq but be sure which one you're using, you could [use a containerized version](https://github.com/openshift/pagerduty-operator/blob/a64ef58978d1427e0956eea390599dab4352eef8/functions.mk#L22).